### PR TITLE
github authentication flow

### PR DESCRIPTION
**Pull request by lizthegrey**
See #315

### CODE REVIEW

**Comment by lizthegrey** [comment]
@benadida do I understand correctly that I could provide a list of github usernames instead of emails for a voter roll once this is enabled, and it should just work?

**Comment by lizthegrey** [comment]
Basically it looks like there's a way to stub people who don't yet have helios accounts with passworded users eg https://github.com/benadida/helios-server/blob/9308a72e2932987051edcf5344315b17b9c6501b/helios/models.py#L864

but there's no way to create a fake `User(user_type='github', user_id=github_username)` when creating voter rolls if I'm reading this correctly

**Comment by lizthegrey** [comment]
https://github.com/benadida/helios-server/blob/9308a72e2932987051edcf5344315b17b9c6501b/helios/models.py#L806 only creates username, password entries. preferences on how I change the bulk upload function to allow uploading `loginmethod:username` user types instead of emails?

**Comment by benadida** [comment]
@lizthegrey you are correct, there's no UX for creating an approved list of Github users as voters. I think it *should* be possible to do by manually creating those voters in the database. Are you planning on using your own install or the official one?

File: helios_auth/auth_systems/github.py [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+  send_mail(
+    subject,
+    body,
+    settings.SERVER_EMAIL,
+    ["%s <%s>" % (user_id, user_info['email'])],
+    fail_silently=False,
+  )
+
+def check_constraint(eligibility, user_info):
+  pass
```
Comment (benadida) [inline helios_auth/auth_systems/github.py:84]: @lizthegrey this hook here could be used to check eligibility against, say, a Github organization or team. Some UX required for sure, or you can write your own `eligibility` data structure manually in the DB. But of course, only if you control the DB. (Clearly, needs work.)

**Review by benadida — APPROVED** [review]
This is great. Approving and merging.

File: helios_auth/auth_systems/github.py (hunk 1 above)
Comment (lizthegrey) [reply helios_auth/auth_systems/github.py:84]: > @lizthegrey you are correct, there's no UX for creating an approved list of Github users as voters. I think it _should_ be possible to do by manually creating those voters in the database. Are you planning on using your own install or the official one?

We were planning to use the official one.

**Pull request merged**